### PR TITLE
comments

### PR DESCRIPTION
**Pull request by p-mayor**
See the comments in this pull request for further feedback.

### CODE REVIEW

File: src/components/menu/Menu.js [hunk 1]
```diff
@@ -61,6 +61,14 @@ function Menu(props) {
...
 }
 
+
+/*
+
+  PGM: I think i wrote the below export. It should be simplified to:
+  export default withAsyncAction("auth", "logout")(Menu)
+
+  */
+
```
Comment (AZimdahl) [inline src/components/menu/Menu.js:71]: I honestly have no idea why this was changed from the original code.

File: src/components/messageFeed/MessageFeed.js [hunk 2]
```diff
@@ -31,6 +31,12 @@ class MessageFeed extends Component {
     let messages = this.props.messages.map(message => {
       return (
         <React.Fragment key={message.id}>
+          {/* 
+          
+          ECS: Consider just passing the entire message object and not passing so many individual values
+          from the same object
+          
+          */}
```
Comment (AZimdahl) [inline src/components/messageFeed/MessageFeed.js:39]: Eric, you wrote this code when showing me how to do the message feed by grabbing the messages without making an unnecessary get request.

File: src/components/signupForm/SignupForm.js [hunk 3]
```diff
@@ -47,11 +47,37 @@ class SignupForm extends Component {
 
     render() {
         const { classes } = this.props;
+
+        /*ECS:
+
+        The onChange attached to the <form> tag should be on each individual <TextField> leaving only
+        the onSubmit on the <form> tag itself
+
```
Comment (AZimdahl) [inline src/components/signupForm/SignupForm.js]: What's the reasoning behind that? Isn't it best to achieve the goal in less code?

File: src/components/signupForm/SignupForm.js [hunk 4]
```diff
@@ -47,11 +47,37 @@ class SignupForm extends Component {
...
+                    
+                        Each textfield controlled by state (controlled components) should have it's
+                        value bound to the associated value in state
+
+                        Example
+                        value={this.state.data.formData.username}
+
+                        See comment above about moving the onchange from the <form> tag to the fields
+                        themselves.
+
```
Comment (AZimdahl) [inline src/components/signupForm/SignupForm.js:73]: This didn't cause problems in this code, but I can see how it wouldve solved so many of my problems in my UserInfo.js

File: src/components/signupForm/SignupForm.js [hunk 5]
```diff
@@ -47,11 +47,37 @@ class SignupForm extends Component {
...
+                        PGM: Also, there is an error in the console coming from the first Textfield component.
+                        It is expecting a boolean for the fullWidth property instead of a string.
+
+                        There's also a warning in the console because you have set the id's for all the Textfield
+                        components to "standard-required" whereas element id's should be unique.
+
```
Comment (AZimdahl) [inline src/components/signupForm/SignupForm.js:79]: Ah, yes, I see the fullwidth issue and you're correct. And you're also right about the standard-required. I should have either changed that to className or removed it altogether. Those were the first things added from Materials-UI so it was the beginning of the learning curve.

File: src/components/userInfo/UserInfo.js [hunk 6]
```diff
@@ -1,3 +1,18 @@
...
+
+
+ECS:
+
+Overall this component seem overly complex especially around state and how the profile image is being
+stored to send over to the API. The mulitple references to the picture in state and the code logic 
+around maintining that make this code hard to follow.
+
+I would also spend some time understaind promises or the alternate syntax of async/await
+
```
Comment (AZimdahl) [inline src/components/userInfo/UserInfo.js:11]: I would agree that the code was hard to follow as well, but it worked from the state it was written, so I didn't want to break it by changing how it was done. I think it would've been best to have the update picture it's own separate component imported into the UserInfo.js

File: src/components/userInfo/UserInfo.js [hunk 7]
```diff
@@ -57,6 +72,16 @@ class UserInfo extends React.Component {
...
   }
 
+  /*
+
+  ECS:
+  
+  Read up on promises for both methods below. getCurrentUserPicture feels overly complex with
+  chained .then() functions. It feels like handleDelete could use a .then() to only logout after the
+  deleteUser completes.
+
```
Comment (AZimdahl) [inline src/components/userInfo/UserInfo.js:82]: Yeah, it would've made more sense to .then the logout after the delete since of the async and needing to be authorized for delete user.

File: src/pages/Home.js [hunk 8]
```diff
@@ -46,6 +47,15 @@ class Home extends React.Component {
         <ListItemText primary="Share what's new on your timeline"/>
       </ListItem>
       </div>
+      
+      {/* ECS: 
+      
+        This type of commented out code/feature that
+        wont be used or is part of an uninished feature
+        should not make it to the master branch. 
+
```
Comment (AZimdahl) [inline src/pages/Home.js:56]: How do you leave out parts of the code in a script while still merging the parts you need to master?

File: src/socialAppService.js [hunk 9]
```diff
@@ -1,3 +1,14 @@
+/*
+
+ECS: lots of repeated code in this file around the headers being attached for auth.
+try an find a way to write this once and reuse it. 
+
+PGM: I like how you clearly broke the methods into sections using comments.
+The getMessagesList method takes in a username argument that it does not use, it should be removed.
+
```
Comment (AZimdahl) [inline src/socialAppService.js:8]: The username argument was added because we expected to be able to only have the messages from a specific user show on a profile, while a message feed page would've shown all messages.

File: src/components/menu/Menu.js (hunk 1 above)
Comment (p-mayor) [reply src/components/menu/Menu.js:71]: Yeah pretty sure I messed this up, honestly I shouldnt have even commented on it. It works, its just inconsistent with the way the other exports are written.

File: src/components/userInfo/UserInfo.js (hunk 6 above)
Comment (p-mayor) [reply src/components/userInfo/UserInfo.js:11]: Yeah, a lot of that mess is my fault. I should have taken the time to clean it up after me and shanita got it working.
